official/keras_application_models/benchmark_main.py

```diff
@@ -0,0 +1,242 @@
+      raise ValueError("Unrecognized callback requested: {}".format(callback))
+
+    if callback_name == model_callbacks.CALLBACKS[0]:
+      exp_per_second_callback = model_callbacks.ExamplesPerSecondCallback(
+          batch_size=batch_size, epoch_size=epoch_size,
+          batch_based=batch_based, epoch_based=epoch_based,
+          metric_logger=metric_logger)
+      callbacks.append(exp_per_second_callback)
+    else:
+      logging_metric_callback = model_callbacks.LoggingMetricCallback(
+          metrics=metrics, metric_logger=metric_logger,
+          batch_based=batch_based, epoch_based=epoch_based)
+      callbacks.append(logging_metric_callback)
+
+  return callbacks
+
+
+def generate_synthetic_input_dataset(image_size, batch_size):
+  """Generate synthetic dataset."""
+  input_shape = (batch_size,) + image_size + (_NUM_CHANNELS,)
+
+  images = tf.zeros(input_shape, dtype=tf.float32)
+  labels = tf.zeros((batch_size, _NUM_CLASSES), dtype=tf.float32)
+
+  return tf.data.Dataset.from_tensors((images, labels)).repeat()
+
+
+def get_default_image_size(model):
+  """Provide default image size for each model."""
+  image_size = (224, 224)
+  if model in ["inception", "xception", "inceptionresnet"]:
+    image_size = (299, 299)
+  elif model in ["nasnetlarge"]:
+    image_size = (331, 331)
+  return image_size
+
+
+def main(_):
+  # Ensure a valid model name was supplied via command line argument
+  if FLAGS.model not in MODELS.keys():
+    raise AssertionError("The --model command line argument should "
+                         "be a key in the `MODELS` dictionary.")
+
+  # Load the model
+  tf.logging.info("Benchmark on {} model...".format(FLAGS.model))
+  keras_model = MODELS[FLAGS.model]
+  model = keras_model(weights=None)
+
+  image_size = get_default_image_size(FLAGS.model)
+  if FLAGS.use_synthetic_data:
+    tf.logging.info("Using synthetic dataset...")
+    batch_size = resnet_run_loop.per_device_batch_size(
+        FLAGS.batch_size, flags_core.get_num_gpus(FLAGS))
+    train_dataset = generate_synthetic_input_dataset(image_size, batch_size)
+    val_dataset = generate_synthetic_input_dataset(image_size, batch_size)
+  else:
+    # Use the actual ImageNet dataset (TODO)
+    pass
+
+  # If run with multiple GPUs
+  num_gpus = flags_core.get_num_gpus(FLAGS)
+  if num_gpus > 0:
+    model = tf.keras.utils.multi_gpu_model(model, gpus=num_gpus)
+
+  # Configure the model
+  model.compile(loss="categorical_crossentropy",
+                optimizer="sgd",
+                metrics=["accuracy"])
+
+  # Create benchmark logger for benchmark logging
+  run_params = {
+      "batch_size": FLAGS.batch_size,
+      "synthetic_data": FLAGS.use_synthetic_data,
+      "benchmark_level": FLAGS.benchmark_level,
+      "train_epochs": FLAGS.train_epochs
+  }
+  benchmark_logger = logger.config_benchmark_logger(FLAGS)
+  benchmark_logger.log_run_info(
+      model_name=FLAGS.model,
+      dataset_name="ImageNet",
```

Comment:
synthetic_data is not real data, so dataset_name should have synthetic as suffix.